tests/tasks/test_api_create_task.py

```diff
@@ -4,9 +4,47 @@
 from app import messages
 from tests.tasks.tasks_base_setup import TasksBaseTestCase
 from tests.test_utils import get_test_request_header
-
+from datetime import timedelta
+from time import sleep
 
 class TestCreateTaskApi(TasksBaseTestCase):
+    # User not involved in mentorship relation tries to create a task (FAIL)
+    # gives 400, UNACCEPTED_STATE_RELATION response
+    def test_create_task_api_not_in_relation(self):
+        auth_header = get_test_request_header(self.first_user.id)
+        expected_response = messages.UNACCEPTED_STATE_RELATION
+        actual_response = self.client.post('/mentorship_relation/%s/task' % self.mentorship_relation_without_first_user.id,
+                                           follow_redirects=True, headers=auth_header, content_type='application/json',
+                                           data=json.dumps(dict(description=self.test_description)))
+        self.assertEqual(400, actual_response.status_code)
+        self.assertDictEqual(expected_response, json.loads(actual_response.data))
+
+    # Valid user tries to create task without description in the body (FAIL)
+    # gives 400, DESCRIPTION_FIELD_IS_MISSING response
+    def test_create_task_api_no_description(self):
+        auth_header = get_test_request_header(self.first_user.id)
+        expected_response = messages.DESCRIPTION_FIELD_IS_MISSING
+        actual_response = self.client.post('/mentorship_relation/%s/task' % self.mentorship_relation_without_first_user.id,
+                                           follow_redirects=True, headers=auth_header, content_type='application/json',
+                                           data=json.dumps(dict()))
+        self.assertEqual(400, actual_response.status_code)
+        self.assertDictEqual(expected_response, json.loads(actual_response.data))
+
+    # Valid user tries to create a task with authentication token expired (FAIL)
+    # gives 401, TOKEN_HAS_EXPIRED response
+    def test_create_task_api_token_expired(self):
+        #generate header that expires in 1 millisecond
+        auth_header = get_test_request_header(
+            self.first_user.id,
+            token_expiration_delta = timedelta(milliseconds = 1)
+        )
+        sleep(2) #sleep till token expires
```

Comment:
Usually Sleep is generally not considered a good practice for tests since it increases the execution time.Try to think of some other way to achieve the same if possible.Anyways yours is a different approach and quite appreciated  :+1: ! Do give this a thought though.